addons/base_import_module/models/ir_module.py

```diff
@@ -106,6 +106,39 @@ def _import_module(self, module, path, force=False):
                     else:
                         IrAttachment.create(values)
 
+        IrAsset = self.env['ir.asset']
+        assets_vals = []
+
+        # Generate 'ir.asset' record values for each asset delared in the manifest
+        for bundle, commands in terp.get('assets', {}).items():
+            for command in commands:
+                directive, target, path = IrAsset._process_command(command)
+                path = path if path.startswith('/') else '/' + path # Ensures a '/' at the start
+                assets_vals.append({
+                    'name': f'{module}.{bundle}.{path}',
+                    'directive': directive,
+                    'target': target,
+                    'path': path,
+                    'bundle': bundle,
+                })
+
+        # Look for existing assets
+        existing_assets = IrAsset.search([('name', 'in', [vals['name'] for vals in assets_vals])])
+        existing_names = existing_assets.mapped('name')
+
+        # Update existing assets
+        for asset in existing_assets:
+            asset.write(next(vals for vals in assets_vals if vals['name'] == asset['name']))
+
+        # Create new assets and attach 'ir.model.data' records to them
+        created_assets = IrAsset.create([vals for vals in assets_vals if vals['name'] not in existing_names])
```

Comment:
That's a completely avoidable `O(m*n)` loop: for each existing asset (m) you're searching through essentially the entire list of `asset_vals` (n). And you're doing that twice.

Instead `existing_assets` should be a map of `name: record`, this way you can iterate on the assets and see whether they exist or not in O(1). And the two dispatch loops can be merged into one: if the asset exists update it, otherwise add it to a list of assets to create.